src/hooks/useGetFeed.ts

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useMemo, useState } from 'react';
+import { toast } from 'react-hot-toast';
+import { API_URL } from '../constants/api';
+import { Stories } from '../types/story';
+
+const POLLING_WAIT = 10000; // ms
+
+export const useGetFeed = () => {
+  const [loading, setLoading] = useState(true);
+  // Page 0 and 1 return the same stores. Start at 0 to allow incrementPage to handle incrementing.
+  const [page, setPage] = useState(0);
+  const [feed, setFeed] = useState<Stories>([]);
+  const [errored, setErrored] = useState(false);
+
+  const incrementPage = async () => {
+    setLoading(true);
+    const nextPage = page + 1;
+    const url = `${API_URL}?page=${nextPage}`;
+
+    try {
+      const response = await fetch(url);
+
+      if (!response.ok) {
+        throw new Error('API Error');
+      }
+
+      const json: Stories = await response.json();
+      // Uncomment to display "new stories" behavoir.
+      // const jsonSliced = json.slice(24);
+      setFeed((feed) => [...feed, ...json]);
+
+      setPage(nextPage); // Only increment page in case of successful call.
+    } catch (error) {
+      console.error('There was an error with the load more API call.'); // Sentry log
+      setErrored(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Retrieve initial feed on load.
+  let initialized = false;
+  useEffect(() => {
+    if (!initialized) {
+      // Alleviate local dev double-load issue.
+      initialized = true; // eslint-disable-line react-hooks/exhaustive-deps
+
+      incrementPage();
+    }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Instantiate polling for new stories on load.
+  useEffect(() => {
+    const timeout = setTimeout(() => {
```

Comment:
As I mentioned, there's modern tools to invoke polling situations. I just figured it was best for me to do this by hand for this test.